sw/airborne/boards/parrot_minidrone/board.c

```diff
@@ -0,0 +1,273 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with paparazzi; see the file COPYING.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * @file boards/parrot_minidrone/board.c
+ *
+ * Parrot Minidrone board initialization functions.
+ *
+ */
+
+/* NOTE: This define is used only for printing the baro type during compilation */
+#ifndef BARO_BOARD
+#define BARO_BOARD BARO_PARROT_MINIDRONE
+#endif
+
+#include <stdlib.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <string.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <pthread.h>
+#include <linux/input.h>
+
+#include <sys/resource.h>// for setrlimit
+#include <errno.h> //Remove if not needed anymore
+
+#include "mcu.h"
+
+// not used atm but thingy below #include <linux/videodev2.h>
+#include "modules/computer_vision/lib/v4l/v4l2.h"
+//#include "peripherals/video_device.h"
+
+#include "boards/parrot_minidrone.h"
+#include "subsystems/electrical.h"
+#include "subsystems/sensors/baro.h"
+#include "subsystems/abi.h"
+
+//By default on a Parrot Minidrone there is no front camera available rherefore bottom and front are set as the same device
+struct video_config_t front_camera = {
+  .output_size = {
+    .w = 640,
+    .h = 480
+  },
+  .sensor_size = {
+    .w = 640,
+    .h = 480
+  },
+  .crop = {
+    .x = 0,
+    .y = 0,
+    .w = 640,
+    .h = 480
+  },
+  .dev_name = "/dev/video0", //TODO start useing the symlink? /dev/vertical_camera
+  .subdev_name = NULL,
+  .format = V4L2_PIX_FMT_YUYV, //AFAIK Sadly no UYUV support
+  .buf_cnt = 60,
+  .filters = 0,
+  .cv_listener = NULL,
+  .fps = 0
+};
+
+struct video_config_t bottom_camera = {
+  .output_size = {
+    .w = 640,
+    .h = 480
+  },
+  .sensor_size = {
+    .w = 640,
+    .h = 480
+  },
+  .crop = {
+    .x = 0,
+    .y = 0,
+    .w = 640,
+    .h = 480
+  },
+  .dev_name = "/dev/video0", //TODO start useing the symlink? /dev/vertical_camera
+  .subdev_name = NULL,
+  .format = V4L2_PIX_FMT_YUYV, //AFAIK Sadly no UYUV support
+  .buf_cnt = 60,
+  .filters = 0,
+  .cv_listener = NULL,
+  .fps = 0
+};
+
+/**
+ * Battery reading thread
+ */
+static void *bat_read(void *data __attribute__((unused)))
+{
+  FILE *fp;
+  char path[16];
+
+  while (TRUE) {
+    /* Open the command for reading. */
+    fp = popen("cat /sys/devices/platform/p6-spi.2/spi2.0/vbat", "r");
+    if (fp == NULL) {
+      printf("Failed to read battery\n");
+    } else {
+      /* Read the output a line at a time - output it. */
+      while (fgets(path, sizeof(path) - 1, fp) != NULL) {
+        int raw_bat = atoi(path);
+        // convert to decivolt
+        // from /bin/mcu_vbat.sh: MILLIVOLTS_VALUE=$(( ($RAW_VALUE * 4250) / 1023 ))
+        electrical.vsupply = (float)((raw_bat * 4250) / 1023) / 1000;
```

Comment:
I guess this needs to be somehow mutexed as this is a threaded function. Consider adding a periodic function to write to the electrical.vsupply with mutex locks.